backend/src/main/java/com/redhat/cloud/notifications/auth/kessel/KesselInventoryAuthorization.java

```diff
@@ -0,0 +1,490 @@
+            )
+            .setRelation(recipientsAuthorizationCriterion.getRelation())
+            .setSubject(
+                SubjectReference.newBuilder()
+                    .setSubject(
+                        ObjectReference.newBuilder()
+                            .setType(ObjectType.newBuilder().setNamespace(KESSEL_RBAC_NAMESPACE).setName(KESSEL_IDENTITY_SUBJECT_TYPE).build())
+                            .setId(getUserId(identity))
+                            .build()
+                    ).build()
+            ).build();
+    }
+
+
+
+    /**
+     * Gets the user identifier from the {@link RhIdentity} object.
+     * @param identity the object to extract the identifier from.
+     * @return the user ID in the format that Kessel expects.
+     */
+    private String getUserId(RhIdentity identity) {
+        return backendConfig.getKesselDomain() + "/" + identity.getUserId();
+    }
+
+    /**
+     * Inventory access checks perform important side effects, so the correct one should be chosen
+     * Checks prior to creates/updates/writes/etc of authorized content should use "update". Otherwise, "check" is fine
+     * for view/read and other non-modifying permission checks.
+     */
+    public enum CheckOperation {
+        CHECK, // normal check: for operations where no modification of the authorized content is intended
+        UPDATE // updating check: check for operations that will modify authorized content in the same transaction
+    }
+
+    CheckOperation getCheckOperation(KesselInventoryResourceType resourceType, KesselPermission permission) {
+        if ((resourceType == KesselInventoryResourceType.WORKSPACE) && (permission instanceof WorkspacePermission workspacePermission)) {
+            return switch (workspacePermission) {
+                case APPLICATIONS_VIEW,
+                     BUNDLES_VIEW,
+                     BEHAVIOR_GROUPS_VIEW,
+                     EVENT_LOG_VIEW,
+                     EVENT_TYPES_VIEW,
+                     INTEGRATIONS_VIEW,
+                     DAILY_DIGEST_PREFERENCE_VIEW
+                    -> CheckOperation.CHECK;
+                case BEHAVIOR_GROUPS_EDIT,
+                     CREATE_DRAWER_INTEGRATION,
+                     CREATE_EMAIL_SUBSCRIPTION_INTEGRATION,
+                     INTEGRATIONS_CREATE,
+                     DAILY_DIGEST_PREFERENCE_EDIT
+                    -> CheckOperation.UPDATE;
+            };
+        } else if ((resourceType == KesselInventoryResourceType.INTEGRATION) && (permission instanceof IntegrationPermission integrationPermission)) {
+            return switch (integrationPermission) {
+                case VIEW, VIEW_HISTORY -> CheckOperation.CHECK;
+                case DELETE, DISABLE, EDIT, ENABLE, TEST -> CheckOperation.UPDATE;
+            };
+        } else {
+            Log.errorf("Resource/permission pair unsupported for Kessel check: %s/%s", resourceType, permission);
+            return CheckOperation.UPDATE;
```

Comment:
Maybe throw an `IllegalArgumentException` here to avoid calling Kessel if we are not able to identify the resource / permission combination.